src/com/twitter/intellij/pants/execution/PantsMakeBeforeRun.java

```diff
@@ -121,6 +126,14 @@ else if (runConfiguration instanceof CommonProgramRunConfigurationParameters) {
     runManagerImpl.setBeforeRunTasks(runConfiguration, Collections.singletonList(pantsMakeTask), false);
   }
 
+  public static void terminatePantsProcess(Project project) {
+    Process process = runningPantsProcesses.get(project);
+    if (process != null) {
+      process.destroy();
+    }
+    runningPantsProcesses.remove(project);
```

Comment:
It'd be safer to use  [remove(Object, Object)](https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ConcurrentHashMap.html#remove(java.lang.Object,%20java.lang.Object))  here